db/db_impl_debug.cc

```diff
@@ -32,6 +32,11 @@ bool DBImpl::TEST_WALBufferIsEmpty() {
   return cur_log_writer->TEST_BufferIsEmpty();
 }
 
+bool DBImpl::TEST_WALBufferIsEmptyNoLock() {
```

Comment:
Perhaps we can overload TEST_WALBufferIsEmpty with a bool lock argument defaulted to true.